### PR TITLE
Fix gzip decompression

### PR DESCRIPTION
**Pull request by Morilli**
`Stream.Read` may read less than the requested number of bytes, which actually does happen in the case of `GZipStream`.

The correct way to handle this is to either copy the stream entirely or use `Read` in a loop until the function returns `0`.

### CODE REVIEW

**Comment by Crauzer** [comment]
What about this:
```cs
internal static class StreamExtensions
{
    public static void ReadUntilDepletion(this Stream stream, Span<byte> buffer)
    {
        int bytesRead = 0;
        do
        {
            bytesRead += stream.Read(buffer[bytesRead..]);
        } while (bytesRead is not 0);

        if (bytesRead != buffer.Length)
            throw new IOException($"Failed to read {buffer.Length} bytes, bytesRead: {bytesRead}");
    }
}
```

File: src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿namespace LeagueToolkit.Utils.Extensions
```
Comment (Crauzer) [inline src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs:1]: Please change to file-scoped namespace

File: src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public static class StreamExtensions
+    {
+        public static void ReadExactly(this Stream stream, Span<byte> buffer)
```
Comment (Crauzer) [inline src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs:5]: change name to `ReadExact`

File: src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs (hunk 1 above)
Comment (Morilli) [reply src/LeagueToolkit/Utils/Extensions/StreamExtensions.cs:1]: This should be configured in the editorconfig accordingly then:
```diff
diff --git a/src/LeagueToolkit/.editorconfig b/src/LeagueToolkit/.editorconfig
index 7eb46fe..c9dc7bb 100644
--- a/src/LeagueToolkit/.editorconfig
+++ b/src/LeagueToolkit/.editorconfig
@@ -109,7 +109,7 @@ csharp_style_prefer_readonly_struct = true
 # Code-block preferences
 csharp_prefer_braces = true
 csharp_prefer_simple_using_statement = true
-csharp_style_namespace_declarations = block_scoped
+csharp_style_namespace_declarations = file_scoped
 csharp_style_prefer_method_group_conversion = true
 csharp_style_prefer_top_level_statements = true

```

**Review by Crauzer — APPROVED** [review]

**Pull request merged**